keras_nlp/layers/rotary_embedding.py

```diff
@@ -16,9 +16,47 @@
 
 
 class RotaryEmbedding(keras.layers.Layer):
-    def __init__(self, rotary_percentage, max_wavelength=10000):
+    """Rotary positional encoding layer.
+
+        Tbjs layer encodes absolute positional information with rotation matrix and naturally
+        incorporates explicit relative position dependency in self-attention formulation.
+        It layer calculates the rotary encoding with a mix of sine and cosine
+        functions with geometrically increasing wavelengths. Defined and formulized
+        in [RoFormer: Enhanced Transformer with Rotary Position Embedding
+    ](https://arxiv.org/abs/2104.09864v4).
+
+        Takes as input the query and key tensors. The input must have shape
+        [batch_size, num_heads, sequence_length, query_length]. This layer will return
```

Comment:
We might want to consider the general form of this. I don't think we want to require a head axis that seems a little two special cased.

We can safely assume that the batch axis is 0, most layers do this. We can also assume feature axis is -1. If we need to take in the sequence dim axis, maybe let's add that as an argument `sequence_axis=1` and allow it to be specified. Then we should test this layer with "multi-head" inputs, and simpler `(batch_size, sequence_length, feature_dim)` inputs.